### PR TITLE
Update to monaco 0.34

### PR DESCRIPTION
**Pull request by CGNonofr**
Also use `monaco-vscode-api` `service-override` feature and do not rely on a custom build of monaco editor (except for the snippets...)

### CODE REVIEW

File: src/tools.ts [hunk 1]
```diff
@@ -1,6 +1,13 @@
 import * as monaco from 'monaco-editor'
 
-function isPasteAction (handlerId: string, payload: unknown): payload is monaco.editor.PastePayload {
+interface PastePayload {
+  text: string
+  pasteOnNewLine: boolean
+  multicursorText: string[] | null
+  mode: string | null
+}
```
Comment (samuel-olivier) [inline src/tools.ts:8]: Why don't we use monaco type anymore ?

File: src/configuration/index.ts [hunk 2]
```diff
@@ -11,7 +11,23 @@ configurationRegistry.registerDefaultConfigurations([{
...
+          fontStyle: 'italic bold'
+        },
+        '*.readonly': {
+          fontStyle: 'bold'
+        },
+        '*.deprecated': {
+          fontStyle: 'strikethrough'
+        }
+      }
+    }
```
Comment (samuel-olivier) [inline src/configuration/index.ts:30]: will we be able to remove those from the themes now that it's here ?

File: src/theme/index.ts [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
+import { setThemes, setDefaultThemes, IThemeExtensionPoint } from 'vscode/service-override/theme'
+import { Disposable } from 'vscode'
 import defaultThemes from '../languages/extensions/themes.json'
 import themeLoader from '../languages/extensions/themeLoader'
 
-const DEFAULT_VSCODE_THEMES: Partial<Record<string, monaco.editor.BuiltinTheme>> = {
-  'Default Light+': 'vs',
-  'Default Dark+': 'vs-dark',
-  'Default High Contrast': 'hc-black'
-  // 'Default High Contrast Light': 'hc-white' // FIXME: Uncomment me with monaco 0.34 which support hc-light theme
```
Comment (samuel-olivier) [inline src/theme/index.ts:12]: does it mean we will have the HC light theme now ?

File: src/types/monaco.d.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { IRange } from 'monaco-editor'
+
+declare module 'monaco-editor' {
+    namespace editor {
+        interface IStandaloneCodeEditor {
+            setHiddenAreas(ranges: IRange[]): void
+        }
+    }
+}
```
Comment (samuel-olivier) [inline src/types/monaco.d.ts:9]: Is this supposed to be temporary ? Is there an issue filed on monaco repo ? If yes maybe add a comment

File: src/tools.ts (hunk 1 above)
Comment (CGNonofr) [reply src/tools.ts:8]: This type is defined internally in vscode, it's not even exported

Our custom build of monaco-editor was exporting it but we don't have it now

File: src/configuration/index.ts (hunk 2 above)
Comment (CGNonofr) [reply src/configuration/index.ts:30]: It's not new, it was just moved, I think we already were able to not have it in the themes

File: src/theme/index.ts (hunk 3 above)
Comment (CGNonofr) [reply src/theme/index.ts:12]: Yes, the bug was in the monaco standalone services

File: src/types/monaco.d.ts (hunk 4 above)
Comment (CGNonofr) [reply src/types/monaco.d.ts]: Nop, it's not temporary, the method was make public by our custom build of monaco

=> https://github.com/CodinGame/monaco-editor-wrapper/pull/57/commits/e02f1a59b79fcb417bcc3b01dd7bb2d094b674f4

**Review by samuel-olivier — APPROVED** [review]

**Pull request merged**